File: trunk/bench/nested_loops.rb
def nested_loops(n)
	x = 0;
	a = 0
	while a<n
		b = 0
		while b<n
			c = 0
			while c<n
				d = 0
				while d<n
					e = 0
					while e<n
						f = 0
						while f<n
							x += 1;
							f += 1
						end
						e += 1
					end
					d += 1
				end
				c += 1
			end
			b += 1
		end
		a += 1
	end
	return x;
end

puts nested_loops(20)

File: bench/nested_loops.lua

function nested_loops(n)
	local x = 0;
	local a = 0
	while a<n do
		local b = 0
		while b<n do
			local c = 0
			while c<n do
				local d = 0
				while d<n do
					local e = 0
					while e<n do
						local f = 0
						while f<n do
							x = x + 1
							f = f + 1
						end
						e = e + 1
					end
					d = d + 1
				end
				c = c + 1
			end
			b = b + 1
		end
		a = a + 1
	end
	return x
end

print(nested_loops(20))
